File: app/src/main/java/com/wealoha/social/beans/CommentDTO.java
package com.wealoha.social.beans;

/**
 * @author:sunkist
 * @see:
 * @since:
 * @description
 * @copyright wealoha.com
 * @Date:2015年3月11日
 */
public class CommentDTO {

	public long createTimeMillis;
	public String userId;
	public boolean mine;
	public String id;
	public String type;
	public String comment;
	/** 回复某人 */
	public String replyUserId;
	public boolean whisper;
	/** 新版评论 */
	public UserDTO user;
	public UserDTO replyUser;

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		PostComment other = (PostComment) obj;
		if (id == null) {
			if (other.getId() != null)
				return false;
		} else if (!id.equals(other.getId()))
			return false;
		return true;
	}
}

File: app/src/main/java/com/wealoha/social/beans/CommentDTO.kt
package com.wealoha.social.beans

/**
 * @author:sunkist
 * @see:
 * @since:
 * @description
 * @copyright wealoha.com
 * @Date:2015年3月11日
 */
data class CommentDTO(
    var createTimeMillis: Long = 0,
    var userId: String,
    var mine: Boolean = false,
    var id: String,
    var type: String? = null,
    var comment: String,

    /** 回复某人  */
    var replyUserId: String,
    var whisper: Boolean = false,

    /** 新版评论  */
    var user: UserDTO,
    var replyUser: UserDTO
) {

    override fun hashCode(): Int {
        val prime = 31
        var result = 1
        result = prime * result + if (id == null) 0 else id.hashCode()
        return result
    }

    override fun equals(obj: Any?): Boolean {
        if (this === obj) return true
        if (obj == null) return false
        if (javaClass != obj.javaClass) return false
        val other = obj as PostComment
        if (id == null) {
            if (other.id != null) return false
        } else if (id != other.id) return false
        return true
    }

    companion object {
        fun fakeForList(toUserDTO: UserDTO, fromUserDTO: UserDTO): CommentDTO {
            return CommentDTO(
                createTimeMillis = 0,
                userId = "",
                mine = false,
                id = "",
                type = null,
                comment = "",
                replyUserId = "",
                whisper = false,
                user = fromUserDTO,
                replyUser = toUserDTO
            )
        }
    }
}